### PR TITLE
No need for else statements

### PR DESCRIPTION
**Pull request by ghost**
When there is a goto statement in an if statement, and then an else statement, that else statement is redundant and creates more jump statements in the resulting asm.

Removing those unneeded else brackets makes the code much more clearer.

### CODE REVIEW

File: src/coreclr/src/pal/src/locale/utf8.cpp [hunk 1]
```diff
@@ -1586,13 +1586,8 @@ class UTF8Encoding
                         charCount--;
                     }
                 }
-                else {
-                    // 2 byte encoding
```
Comment (jkotas) [inline src/coreclr/src/pal/src/locale/utf8.cpp]: Removing this comment does not make the code easier to understand. The extra else and scope are here on purpose to make the code easier to understand.

**Comment by jkotas** [comment]
> Removing those unneeded else brackets makes the code much more clearer.

That is not correct.

**Comment by janvorli** [comment]
> When there is a goto statement in an if statement, and then an else statement, that else statement is redundant and creates more jump statements in the resulting asm.

This really depends on how the compiler optimizes the code. I would not be surprised if there was no difference at all in the code generated with and without your change.

**Comment by jkotas** [comment]
Duplicate of https://github.com/dotnet/runtime/pull/32402

**Pull request closed without merging**